lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -493,8 +502,12 @@ def post(self):
             # be interpreted as formatting commands.
             self.logger.warning("{}", str(e))
             abort(400, message=str(e))
+        except UnVerifiedUser as e:
+            self.logger.warning("{}", str(e))
+            abort(401, message=str(e))
         return self._call(requests.post, new_data)
 
+    @Auth.token_auth.login_required()
```

Comment:
Uh oh. `/controllers/months` wasn't intended to be authenticated. There's nothing user-specific here, and it needs to be able to return the range of index months for unauthenticated calls (public data).

However, this raises an interesting question, because perhaps it _should_ be authenticated rather than allowing just anyone to determine the full range of in-use run index months? E.g., maybe instead of just looking for all the indices and filtering out the date-stamped run indices with the right prefix (as it currently does), we _could_ run a more complex query to find all run indices which have run documents owned by the client or all run indices which have public documents. This, for example, wouldn't give a recent user a range of years in which they wouldn't possibly own any data. Hmm.

However, with the change to allow for an unauthorized query (for public data), let's leave it this way. I'll pull out the comment about user-constrained month ranges (which I think is a good idea anyway) into an issue and we'll worry about that aspect later.

Reply:
@dbutenhof, can you add a reference to that tracking issue here?

Reply:
I made the authentication optional